### PR TITLE
Remove redundant side macros/funcs

### PR DESCRIPTION
**Pull request by gruxor**
## Description
Replaces GET_BATTLER_SIDE and GET_BATTLER_SIDE2 with the existing GetBattlerSide, which is now inlined. Also changes all instances of GetBattlerPosition with direct gBattlerPositions usage.

## **Discord contact info**
crater.

### CODE REVIEW

**Review by mrgriffin — CHANGES_REQUESTED** [review]

**Comment by DizzyEggg** [comment]
Is removing GetBattlerPosition a good idea? I think it's more readable the way it is now. 
Also, if someone modified that function to behave differently, it's gonna be a problem.

I would personally make it an inline function like the rest.

**Comment by gruxor** [comment]
> Is removing GetBattlerPosition a good idea? I think it's more readable the way it is now. Also, if someone modified that function to behave differently, it's gonna be a problem.
> 
> I would personally make it an inline function like the rest.

Re-added and applied everywhere, only uses of `gBattlerPositions` should be on assignments now. Mostly regex'd and then proofread by the compiler, so could use a manual look-through if someone wants to check over the changed instances.

**Comment by mrgriffin** [comment]
I can try to review this later in the weekend/next week. If somebody else wants to get to it first they're more than welcome to and I can change my review to approve.

File: include/battle.h [hunk 1]
```diff
@@ -1012,4 +998,25 @@ extern u8 gLastUsedBall;
...
+}
+
+static inline struct Pokemon *GetSideParty(u32 side)
+{
+    return (side == B_SIDE_PLAYER) ? gPlayerParty : gEnemyParty;
+}
+
+static inline struct Pokemon *GetBattlerParty(u32 battlerId)
+{
+    extern u8 GetBattlerSide(u8 battler);
```
Comment (DizzyEggg) [inline include/battle.h:1018]: Why is this extern needed here?

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Other than the weird extern and conflicts, I think this should be good to go.

**Comment by DizzyEggg** [comment]
Replaced by #3299

**Pull request closed without merging**